src/lib/constants.js

```diff
@@ -14,12 +14,16 @@ const constants = {
   "REQUEST_SEARCH_SUGGESTIONS",
   "RECEIVE_SEARCH_SUGGESTIONS",
   "UPDATE_SEARCH_STRING",
+  "RECEIVE_UPDATE_SITES",
   "REQUEST_CURRENT_SEARCH_ENGINE",
   "RECEIVE_CURRENT_SEARCH_ENGINE",
   "REQUEST_VISIBLE_SEARCH_ENGINES",
   "RECEIVE_VISIBLE_SEARCH_ENGINES",
   "REQUEST_FRECENT",
-  "RECEIVE_FRECENT"
+  "RECEIVE_FRECENT",
+  "REQUEST_INIT_USER_DATABASE",
+  "REQUEST_INIT_USER_DATABASE_COMPLETE",
+  "REQUEST_UNDO_DIALOG_VISIBILITY"
 ].forEach(action => constants[action] = action);
```

Comment:
It is the suggested pattern by Redux, presumably to avoid accidental misspelling (redux warns if you try to pass an action without a `type`... however, I don't love it.

I am playing around with an idea for an "action manager" that might be a little better